var/spack/repos/builtin/packages/samrai/package.py

```diff
@@ -53,24 +55,26 @@ class Samrai(Package):
     depends_on("zlib")
     depends_on("hdf5+mpi")
     depends_on("boost")
+    depends_on("m4")
```

Comment:
See http://spack.readthedocs.io/en/latest/packaging_guide.html#dependency-specs

`build` types need to be present during the build, but can be deleted afterwards and the installation will still work. The dependency is put in the PATH.

`link` types need to be present during the build and after the installation. If you delete the dependency, the installation would no longer work, so Spack prevents you from doing that. The dependency is added to the RPATH, `-L`, and `-I`.

`run` types don't need to be present during the build, but the installation won't work unless the dependency is installed. Spack prevents you from uninstalling these dependencies.

You can combine these dependencies as needed in a tuple. The default is `('build', 'link')`.